Frontend/ui-driver/src/Flow.purs

```diff
@@ -2833,7 +2840,49 @@ updateBannerAndPopupFlags = do
                 { autoPayBanner = autopayBannerType
                 , subscriptionPopupType = subscriptionPopupType
                 , waitTimeStatus = RC.waitTimeConstructor $ getValueToLocalStore WAITING_TIME_STATUS
+                , showCoinsPopup = getValueToLocalStore VISITED_DRIVER_COINS_PAGE /= "true"
                 }
               }
         )
 
+
+
+driverEarningsFlow :: FlowBT String Unit
+driverEarningsFlow = do 
+  (GlobalState globalState) <- getState
+  appConfig <- getAppConfig Constants.appConfig
+  let earningScreenSate = globalState.driverEarningsScreen
+  if not earningScreenSate.data.anyRidesAssignedEver
+  then do
+    profileSummaryResponse <-  lift $ lift $ Remote.driverProfileSummary ""
+    let anyRidesAssignedEver = case profileSummaryResponse of 
```

Comment:
try to use `either` function instead of case matching

https://pursuit.purescript.org/packages/purescript-either/6.1.0/docs/Data.Either#v:either